src/jbuild.ml

```diff
@@ -248,6 +252,18 @@ module Preprocess = struct
   let pps = function
     | Pps { pps; _ } -> pps
     | _ -> []
+
+  let filter_optional spec pp =
+    let disable_optional pp =
+      if Pp.is_optional pp && not (String_set.mem (Pp.to_string pp) spec) then None
+      else Some(pp)
+    in
+    match pp with
+    | Pps { pps; flags } ->
+      let pps = List.filter_map pps ~f:disable_optional in
+      if pps = [] then No_preprocessing
```

Comment:
Does it really make sense to have flags but no actual preprocessor?  Would this be for the driver?